src/libs/viewpoints/viewpoint_extractor.ts

```diff
@@ -0,0 +1,536 @@
+  }
+
+  /**
+   * This methor will run the patter viewpoint extrator.
+   * 
+   * @param id The name of viewpoint extractor 
+   * @param pattern The viewpoint extractor approach based on OntoUML Patterns
+   * @returns A viewpoint based on a pattern choosen
+   */
+  extractViewpoint(id: string, pattern: string): Viewpoint {
+    let viewpoint = new Viewpoint('Viewpoint of ' + pattern);
+
+    switch (pattern) {
+      case ClassStereotype.KIND: viewpoint = this.kindView();
+        break;
+      case ClassStereotype.SUBKIND: viewpoint = this.subKindView();
+        break;
+      case ClassStereotype.ROLE: viewpoint = this.roleView();
+        break;
+      case ClassStereotype.PHASE: viewpoint = this.phaseView();
+        break;
+      case ClassStereotype.RELATOR: viewpoint = this.relatorView();
+        break;
+      case ClassStereotype.MODE: viewpoint = this.modeView();
+        break;
+      case 'NON_SORTAL': viewpoint = this.nonSortalView();
+        break;
+    }
+
+
+    return viewpoint;
+  }
+
+  /**
+   * This method filter all Kinds and related relations ando group them in a view
+   * 
+   * @returns A viewpoint extracted using the Kind Pattern
+   */
+  kindView(): Viewpoint {
+    const viewpoint = new Viewpoint('Kind View');
+
+    /**Step 1 Get all Kinds in model
+     * 
+     * Step 2 Get all relations between Kinds
+     * 
+     * Step 3 Get all generalizations between Kinds
+     */
+
+    const classes = this.project.getAllClassesByStereotype(ClassStereotype.KIND);
+
+    viewpoint.addClasses(classes);
+    viewpoint.addRelations(this.getRelationsBetweenClasses(classes));
+    viewpoint.addGeneralizations(this.getGeneralizationsBetweenClasses(classes));
+
+    viewpoint.removeDuplicates();
+
+    return viewpoint;
+  }
+
+  /**
+     * This method filter all Subkinds, respective sortal ancestors  and related relations and group them in a view
+     * 
+     * @returns A viewpoint extracted using the Subkind Pattern
+     */
+  subKindView(): Viewpoint {
+    const viewpoint = new Viewpoint('SubKind View');
+
+    /**Step 1 Get all Subkinds in model
+     * 
+     * Step 2 Get all sortal ancestors 
+     * 
+     * Step 3 Get all relations between all classes (subkinds and sortal ancestors)
+     * 
+     * Step 4 Get all generalizations between all classes (subkinds and sortal ancestors)
+     */
+
+    const classes = this.project.getAllClassesByStereotype(ClassStereotype.SUBKIND);
+
+    let generals = [];
+    for (let i = 0; i < classes.length; i++) {
+      generals = generals.concat(classes[i].getAncestors().filter((cl) => cl.hasSortalStereotype()));
+    }
+
+    viewpoint.addClasses(classes);
+    viewpoint.addClasses(generals);
+
+    viewpoint.removeDuplicates();
+
+    viewpoint.addRelations(this.getRelationsBetweenClasses(viewpoint.classes));
+    viewpoint.addGeneralizations(this.getGeneralizationsBetweenClasses(viewpoint.classes));
+
+    viewpoint.removeDuplicates();
+
```

Comment:
Aren't generalization sets included in the subkind view?

Reply:
We didn't specify this. At least the generalization must be. I dont't see problem considering generalizationset